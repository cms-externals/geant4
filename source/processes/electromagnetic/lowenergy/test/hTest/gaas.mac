#=======================================================
# Macro file "gaas.mac" for the GaAs test by hTest.cc
# 10.04.01 V.Ivanchenko
#======================================================= 
/control/verbose           2
/run/verbose               2
/step/verbose              0
/tracking/verbose          0
/stepping/verbose          0
/event/verbose             0
/hTest/verbose             0
#
/hTest/physics/cutGamma    0.001 mm
/hTest/physics/cutElectron 0.001 mm
/hTest/physics/MaxStep     0.1 mm
/hTest/physics/EMList      LowEnergy
/hTest/physics/HadronList  none
/hTest/AbsorberMaterial    GaAs
/hTest/AbsorberThick       0.001 mm
/hTest/NumberOfAbsorbers   60
/hTest/WorldMaterial       Vacuum
/hTest/numberAbsToSave     0
/hTest/HistoName           gaas.paw 
/hTest/NumberOfEvents      1000
/hTest/HistoNumber         0
#
/gun/particle              proton
/gun/energy                1.0 MeV

