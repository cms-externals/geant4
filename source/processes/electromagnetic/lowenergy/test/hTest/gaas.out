**********************************************
 Geant4 version 
                                (06-Apr-2001)
             Copyright : Geant4 Collaboration
**********************************************
Detector Construction is defined
HsPhysicsList: <LowEnergy> EMPhysicsList is set
HsPhysicsList: <none> HadronicPhysicsList is set
Physics List is defined
User actions will be initialized
hTestPrimaryGeneratorMessenger: Construct 
hTestPrimaryGeneratorAction is defined
hTestRunAction is defined
EventAction is defined
TrackingAction is defined
UI interface is started
#=======================================================
# Macro file "gaas.mac" for the GaAs test by hTest.cc
# 10.04.01 V.Ivanchenko
#======================================================= 
/run/verbose               2
/step/verbose              0
/tracking/verbose          0
/stepping/verbose          0
/event/verbose             0
/hTest/verbose             0
#
/hTest/physics/cutGamma    0.001 mm
/hTest/physics/cutElectron 0.001 mm
/hTest/physics/MaxStep     0.1 mm
hTestPhysicsList: MaxChargedStep = 0.1 mm
/hTest/physics/EMList      LowEnergy
/hTest/physics/HadronList  none
/hTest/AbsorberMaterial    GaAs
/hTest/AbsorberThick       0.001 mm
/hTest/NumberOfAbsorbers   60
/hTest/WorldMaterial       Vacuum
/hTest/numberAbsToSave     0
/hTest/HistoName           gaas.paw 
/hTest/NumberOfEvents      1000
/hTest/HistoNumber         0
#
/gun/particle              proton
/gun/energy                1.0 MeV

Start initialisation
userDetector->Construct() start.

***** Table : Nb of materials = 18 *****

 Material: Beryllium            density:  1.848 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  35.267 cm
   --->  Element: Beryllium    Z =  4.0   N =   9.0   A =   9.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Aluminum            density:  2.700 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   8.893 cm
   --->  Element: Aluminum    Z = 13.0   N =  27.0   A =  26.98 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Carbon            density:  2.000 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  21.348 cm
   --->  Element: Carbon    Z =  6.0   N =  12.0   A =  12.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Silicon            density:  2.330 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   9.368 cm
   --->  Element: Silicon    Z = 14.0   N =  28.1   A =  28.09 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: liquidArgon            density:  1.390 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  14.065 cm
   --->  Element: liquidArgon    Z = 18.0   N =  40.0   A =  39.95 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Iron            density:  7.870 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.759 cm
   --->  Element: Iron    Z = 26.0   N =  55.9   A =  55.85 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Copper            density:  8.960 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.436 cm
   --->  Element: Copper    Z = 29.0   N =  63.5   A =  63.55 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Gold            density: 19.320 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   3.344 mm
   --->  Element: Gold    Z = 79.0   N = 197.0   A = 196.97 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Lead            density: 11.350 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   5.612 mm
   --->  Element: Lead    Z = 82.0   N = 207.2   A = 207.19 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Water        H_2O   density:  1.000 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  36.092 cm
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  11.21 %  Abundance  66.67 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  88.79 %  Abundance  33.33 %

 Material: Methane        CH_4   density:  0.667 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 696.375 m
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  25.19 %  Abundance  80.00 %
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass:  74.81 %  Abundance  20.00 %

 Material: Graphite        Graphite   density:  2.265 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  18.834 cm
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: GaAs        GaAs   density:  5.318 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   2.293 cm
   --->  Element: GalliumGa   Z = 31.0   N =  69.7   A =  69.72 g/mole  fractionMass:  48.20 %  Abundance  50.00 %
   --->  Element: ArsenicumAs   Z = 33.0   N =  74.9   A =  74.92 g/mole  fractionMass:  51.80 %  Abundance  50.00 %

 Material: Ethane        C_2H_6   density: 424.100 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.076 m
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  20.16 %  Abundance  75.00 %
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass:  79.84 %  Abundance  25.00 %

 Material: CsI        CsI   density:  4.530 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.852 cm
   --->  Element: CesiumCs   Z = 55.0   N = 132.9   A = 132.91 g/mole  fractionMass:  51.15 %  Abundance  50.00 %
   --->  Element: IodideI   Z = 53.0   N = 126.9   A = 126.90 g/mole  fractionMass:  48.85 %  Abundance  50.00 %

 Material: Air            density:  1.290 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 285.161 m
   --->  Element: NitrogenN   Z =  7.0   N =  14.0   A =  14.01 g/mole  fractionMass:  70.00 %  Abundance  72.71 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  30.00 %  Abundance  27.29 %

 Material: Mylar            density:  1.390 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  28.868 cm
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass:  55.00 %  Abundance  30.30 %
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:   8.33 %  Abundance  54.55 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  36.67 %  Abundance  15.15 %

 Material: Vacuum            density:  0.000 mg/cm3  temperature:   2.73 K  pressure:   0.00 atm  RadLength: 204727576.737 pc
   --->  Element: Vacuum    Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %


The  WORLD   is made of  of Vacuum. The transverse size (XY) of the world is 1e+02 mm
The ABSORBER is made of 60 items of 0.001 mm of GaAs. The transverse size (XY) is 1e+02 mm
physicsList->Construct() start.
LowEnergy Electromagnetic PhysicsList is initilized
LowE EM processes for GenericIon
LowE GenericIon
LowE EM processes for He3
LowE He3
LowE EM processes for IonAr40
LowE IonAr40
LowE EM processes for IonC12
LowE IonC12
LowE EM processes for IonFe56
LowE IonFe56
LowE EM processes for alpha
LowE alpha
LowE EM processes for anti_kaon0
LowE EM processes for anti_lambda
LowE EM processes for anti_neutron
LowE EM processes for anti_nu_e
LowE EM processes for anti_nu_mu
LowE EM processes for anti_omega-
LowE EM processes for anti_proton
LowE anti_proton
LowE EM processes for anti_sigma+
LowE EM processes for anti_sigma-
LowE EM processes for anti_sigma0
LowE EM processes for anti_xi-
LowE EM processes for anti_xi0
LowE EM processes for chargedgeantino
LowE EM processes for deuteron
LowE deuteron
LowE EM processes for e+
LowE e+
LowE EM processes for e-
LowE e-
>>>>>>>>> New G4LowEnergyIonisation 0x892ba40
LowE EM processes for eta
LowE EM processes for eta_prime
LowE EM processes for gamma
LowE gamma
LowE EM processes for geantino
LowE EM processes for kaon+
LowE kaon+
LowE EM processes for kaon-
LowE kaon-
LowE EM processes for kaon0
LowE EM processes for kaon0L
LowE EM processes for kaon0S
LowE EM processes for lambda
LowE EM processes for mu+
LowE mu+
MuonMinusCaptureAtRest is created 
LowE EM processes for mu-
LowE mu-
MuonMinusCaptureAtRest is created 
LowE EM processes for neutron
LowE EM processes for nu_e
LowE EM processes for nu_mu
LowE EM processes for omega-
LowE EM processes for pi+
LowE pi+
LowE EM processes for pi-
LowE pi-
LowE EM processes for pi0
LowE EM processes for proton
LowE proton
LowE EM processes for sigma+
LowE sigma+
LowE EM processes for sigma-
LowE sigma-
LowE EM processes for sigma0
LowE EM processes for triton
LowE triton
LowE EM processes for xi-
LowE EM processes for xi0
physicsList->setCut() start.
hTestPhysicsList: Set cuts for all particles 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV to 1e+02 TeV in 100 bins. 
*** LEion using Tcut =0.00048
*** LEion using Tcut =0.00082
*** LEion using Tcut =0.00057
*** LEion using Tcut =0.00066
*** LEion using Tcut =0.00026
*** LEion using Tcut =0.0036
*** LEion using Tcut =0.0042
*** LEion using Tcut =0.0088
*** LEion using Tcut =0.0044
*** LEion using Tcut =0.0002
*** LEion using Tcut =1e-05
*** LEion using Tcut =0.00069
*** LEion using Tcut =0.0019
*** LEion using Tcut =5.5e-05
*** LEion using Tcut =0.0013
*** LEion using Tcut =1e-05
*** LEion using Tcut =0.00034
*** LEion using Tcut =1e-05

LowEnergyIoni:  Low energy ionisation code with implementation of the continuous dE/dx part.
 At present it can be used for electrons  in the energy range [250eV,100GeV]
 the process must work with G4LowEnergyBremsstrahlung
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Beryllium
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Aluminum
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Carbon
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Silicon
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material liquidArgon
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Iron
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Copper
*** LE Bremsstrahlung using Gamma Tcut = 0.0017 for material Gold
*** LE Bremsstrahlung using Gamma Tcut = 0.0012 for material Lead
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Water
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Methane
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Graphite
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material GaAs
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Ethane
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material CsI
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Air
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Mylar
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Vacuum

LowEnBrem:  Total cross sections from EEDL database,Gamma energy sampled from a parametrised formula.Implementation of the continuous dE/dx part.
 At present it can be used for electrons  in the energy range [250eV,100GeV]
 the process must work with G4LowEnergyIonisation

annihil:  Total cross section from Heilter formula (annihilation into 2 photons).
        gamma energies sampled according Heitler
        PhysicsTables from 10 keV to 10 TeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV to 1e+02 TeV in 100 bins. 

MuIoni:  knock-on electron cross sections .
          Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
      PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 

MuBrems:  theoretical cross section 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 

MuPairProd:  theoretical cross sections 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV to 1e+02 TeV in 100 bins. 

hLowEIoni:    Knock-on electron cross sections . 
        Good description above the mean excitation energy.
        Delta ray energy sampled from  differential Xsection.
        PhysicsTables from 10 eV  to 1e+02 TeV  in 200 bins.
        Electronic stopping power model is  ICRU_R49p
        from 1 keV  to 2 MeV .

        Parametrisation model for antiprotons is  ICRU_R49p
        from 50 keV  to 2 MeV .
        Parametrization of the Barkas effect is switched on.
        Nuclear stopping power model is ICRU_R49
           material       min.delta energy(keV) 

Methane          0.043
Air          0.086
Vacuum          0.021
============= The cut Energy ==============================
                      gamma            e-           
Cut in range                  1 mum             1 mum
Cut in energy
 Beryllium                           10 eV           484 eV
   Aluminum                           10 eV           820 eV
   Carbon                           10 eV           570 eV
   Silicon                           10 eV           656 eV
   liquidArgon                           10 eV           259 eV
   Iron                           10 eV          3.56 keV
   Copper                           10 eV          4.15 keV
   Gold                         1.72 keV          8.77 keV
   Lead                         1.21 keV          4.44 keV
   Water                           10 eV           202 eV
   Methane                           10 eV            10 eV
   Graphite                           10 eV           691 eV
   GaAs                           10 eV          1.93 keV
   Ethane                           10 eV          54.8 eV
   CsI                           10 eV          1.33 keV
   Air                           10 eV            10 eV
   Mylar                           10 eV           339 eV
   Vacuum                           10 eV            10 eV
===================================================
Start event loop for 1000 events
### Run 0 start
hTestHisto: Histograms will be saved to the file <gaas.paw>
Start closing geometry.
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=4.7s Real=4.7s Sys=0.02s
hTestRunAction: End of run actions are started
hTestHisto: End of run actions are started
========================================================
Range(mm)= 0.01169; Stragling(mm)= 0.0002494 +- 7.9e-06
========================================================
Histograms and Ntuples are saved
Second mac file is applied
/control/execute gaas1.mac
#=======================================================
# Macro file "gaas1.mac" for the GaAs test by hTest.cc
# 10.04.01 V.Ivanchenko
#======================================================= 
/gun/energy                1.2 MeV
/run/beamOn                1000
### Run 1 start
hTestHisto: Histograms will be saved to the file <gaas.paw>
 *** WARNING in HBOOK1 : Already existing histogram replaced : ID=       1 
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=6.4s Real=6.5s Sys=0.04s
hTestRunAction: End of run actions are started
hTestHisto: End of run actions are started
========================================================
Range(mm)= 0.01507; Stragling(mm)= 0.0003613 +- 1.1e-05
========================================================
Histograms and Ntuples are saved
/gun/energy                1.4 MeV
/run/beamOn                1000
### Run 2 start
hTestHisto: Histograms will be saved to the file <gaas.paw>
 *** WARNING in HBOOK1 : Already existing histogram replaced : ID=       1 
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=8.5s Real=8.6s Sys=0.05s
hTestRunAction: End of run actions are started
hTestHisto: End of run actions are started
========================================================
Range(mm)= 0.01878; Stragling(mm)= 0.0003841 +- 1.2e-05
========================================================
Histograms and Ntuples are saved
/gun/energy                1.6 MeV
/run/beamOn                1000
### Run 3 start
hTestHisto: Histograms will be saved to the file <gaas.paw>
 *** WARNING in HBOOK1 : Already existing histogram replaced : ID=       1 
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=11s Real=11s Sys=0.07s
hTestRunAction: End of run actions are started
hTestHisto: End of run actions are started
========================================================
Range(mm)= 0.02279; Stragling(mm)= 0.0004696 +- 1.5e-05
========================================================
Histograms and Ntuples are saved
/gun/energy                1.8 MeV
/run/beamOn                1000
### Run 4 start
hTestHisto: Histograms will be saved to the file <gaas.paw>
 *** WARNING in HBOOK1 : Already existing histogram replaced : ID=       1 
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=13s Real=14s Sys=0.08s
hTestRunAction: End of run actions are started
hTestHisto: End of run actions are started
========================================================
Range(mm)= 0.02708; Stragling(mm)= 0.0005178 +- 1.6e-05
========================================================
Histograms and Ntuples are saved
/gun/energy                2.0 MeV
/run/beamOn                1000
### Run 5 start
hTestHisto: Histograms will be saved to the file <gaas.paw>
 *** WARNING in HBOOK1 : Already existing histogram replaced : ID=       1 
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=16s Real=16s Sys=0.16s
hTestRunAction: End of run actions are started
hTestHisto: End of run actions are started
========================================================
Range(mm)= 0.03172; Stragling(mm)= 0.0006751 +- 2.1e-05
========================================================
Histograms and Ntuples are saved
/gun/energy                2.2 MeV
/run/beamOn                1000
### Run 6 start
hTestHisto: Histograms will be saved to the file <gaas.paw>
 *** WARNING in HBOOK1 : Already existing histogram replaced : ID=       1 
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=18s Real=18s Sys=0.14s
hTestRunAction: End of run actions are started
hTestHisto: End of run actions are started
========================================================
Range(mm)= 0.03659; Stragling(mm)= 0.0008632 +- 2.7e-05
========================================================
Histograms and Ntuples are saved
/gun/energy                2.4 MeV
/run/beamOn                1000
### Run 7 start
hTestHisto: Histograms will be saved to the file <gaas.paw>
 *** WARNING in HBOOK1 : Already existing histogram replaced : ID=       1 
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=20s Real=20s Sys=0.1s
hTestRunAction: End of run actions are started
hTestHisto: End of run actions are started
========================================================
Range(mm)= 0.04181; Stragling(mm)= 0.000826 +- 2.6e-05
========================================================
Histograms and Ntuples are saved
runManager will be deleted
G4 kernel has come to Quit state.
Deletion of G4 kernel class start.
UserDetectorConstruction deleted.
UserPhysicsList deleted.
UserRunAction deleted.
UserPrimaryGenerator deleted.
G4SDManager deleted.
EventManager deleted.
UImanager deleted.
StateManager deleted.
RunManager is deleting.
runManager is deleted
